Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +128,310 @@ public static class CustomEntitySearch
+                                    break;
+                            }
+                        }
+                        if (currTextCharIndex >= textCharArray.Length || 
+                            (currWordCharIndex >= wordCharArray.Length && textCharArray[currTextCharIndex].IsDelineating()))
+                        {
+                            if (wordCharArray.Length - currWordCharIndex > 0)
+                                currMismatch += wordCharArray.Length - currWordCharIndex;
+                            int initialOffsetIndex = currTextCharIndex - wordFound.ToString().Length;
+                            if (currMismatch <= leniency)
+                            {
+                                if (bestMismatchPreWhitespace > -1 && bestMismatchPreWhitespace < currMismatch)
+                                {
+                                    entities.Add(
+                                    new Entity
+                                    {
+                                        Category = "customEntity",
+                                        Value = wordFound.GetStringBuilder().Remove(whitespaceOffset - initialOffsetIndex - 1, (int)bestMismatchPreWhitespace).ToString(),
+                                        Offset = initialOffsetIndex,
+                                        Confidence = leniency - bestMismatchPreWhitespace - (wordCharArray.Length - currWordCharIndex)
+                                    });
+                                    currTextCharIndex = whitespaceOffset;
+                                }
+                                else
+                                {
+                                    entities.Add(
+                                    new Entity
+                                    {
+                                        Category = "customEntity",
+                                        Value = wordFound.ToString(),
+                                        Offset = initialOffsetIndex,
+                                        Confidence = leniency - currMismatch - (wordCharArray.Length - currWordCharIndex)
+                                    });
+
+                                    if (bestMismatchPreWhitespace > -1 && whitespaceOffset != initialOffsetIndex)
+                                    {
+                                        currTextCharIndex = whitespaceOffset;
+                                    }
+                                }
+
+                                if (!entitiesFound.Contains(checkMatch))
+                                    entitiesFound.Add(checkMatch);
+                                currWordCharIndex = 0;
+                                wordFound.GetStringBuilder().Clear();
+                                currMismatch = 0;
+                                bestMismatchPreWhitespace = -1;
+                            }
+                            else
+                                currMismatch = leniency + 1;
```

Comment:
do you need this else statement? it looks like leniency is already > currmismatch to hit this?